### PR TITLE
Registry integration test: Recursion level

### LINKED ISSUE #895: Registry integration tests: Recursion level
## Description

After the development of wazuh/wazuh#5576, new integration tests are going to be needed.
Each file may perform several tests. The structure of the directory will be:

### test_recursion_level

| Test                 | Action                                          | Expected result                                             |
| -------------------- | ----------------------------------------------- | ----------------------------------------------------------- |
| test_recursion_level | Monitor registry specifying the recursion level | Registries over the recursion level should not be monitored |

### PR DESCRIPTION
**Pull request by antoniomanuelfr**
Hello team,

This PR aims to add a new test for the windows registry integration tests
Closes #895

# Tests logic
This test has only one case: 
- The test will monitor some keys with different recursion level. Then it creates values in the lower levels and in the last levels and check that the alerts are generated.
- After that, the test creates keys with a higher recursion level and checks that no events are generated in these keys

# Tests checks

- [X] Proven that tests **pass** when they have to pass
- [X] Proven that tests **fail** when they have to fail
- [ ] Tested and passed in Jenkins. Build URL: `<YOUR_JENKINS_BUILD_URL>`

## Windows

- When the expected behavior occurs:

```
python -m pytest -vvx test_recursion_level\
=================================== test session starts ============
metadata: {'Python': '3.7.3', 'Platform': 'Windows-7-6.1.7601-SP1',
collected 4 items

test_recursion_level_registry.py PASSED [ 25%]
test_recursion_level_registry.py PASSED [ 50%]
test_recursion_level_registry.py PASSED [ 75%]
test_recursion_level_registry.py PASSED [100%]

=========================================== 4 passed in 173.19s (0:02:53) ==============

```
Best regards.

### CODE REVIEW

File: tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py [hunk 1]
```diff
@@ -3,17 +3,14 @@
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 import os
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py:4]: Please, add one blank line between the comment and the first import.
```suggestion

import os
```

File: tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py [hunk 2]
```diff
@@ -22,152 +19,134 @@
...
+
+               'REGISTRY_1': os.path.join(key, registry_recursion_1),
+               'LEVEL_1': rl_dict[registry_recursion_1],
+
+               'REGISTRY_2': os.path.join(key, registry_recursion_5),
+               'LEVEL_2': rl_dict[registry_recursion_5],
+
+               'REGISTRY_3': os.path.join(key, registry_recursion_29),
+               'LEVEL_3': rl_dict[registry_recursion_29]
+}
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py:49]: This closing bracket does not match visual indentation.
```suggestion
               }
```

File: tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py [hunk 3]
```diff
@@ -22,152 +19,134 @@
...
-def recursion_test(key, registry, subkey, recursion_level, timeout=1, edge_limit=2, ignored_levels=1, is_scheduled=False):
+    for reg, rl in rl_dict.items():
+        path = str(reg)
+        for n in range(int(rl)):
+            path = os.path.join(path, '' + str(n + 1))
+        create_registry(registry_parser[key], path, KEY_WOW64_64KEY)
+
+# Tests
+
+@pytest.mark.parametrize('root_key, registry, arch, edge_limit, ignored_levels', [
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py:76]: Please, add two blank lines before a function declaration.
```suggestion

@pytest.mark.parametrize('root_key, registry, arch, edge_limit, ignored_levels', [
```

File: tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py [hunk 4]
```diff
@@ -22,152 +19,134 @@
...
+                       min_timeout=global_parameters.default_timeout)
+
+    path_list = list()
+    # For recursion lower levels, execute registry_value_cud in every level.
+    if recursion_level < edge_limit:
+        for level in range(recursion_level):
+            path = os.path.join(path, str(level + 1))
+            path_list.append(path)
+    else:
+        for level in range (recursion_level):
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py:138]: Remove the whitespace before the `(`.
```suggestion
        for level in range(recursion_level):
```

File: tests/integration/test_fim/test_registry/test_recursion_level/data/wazuh_recursion_windows_registry.yaml [hunk 5]
```diff
@@ -7,22 +8,18 @@
...
-        - recursion_level: "5"
-        - FIM_MODE
+        - recursion_level: LEVEL_2
     - windows_registry:
-        value: "HKEY_LOCAL_MACHINE\\SOFTWARE\\recursion512"
+        value: REGISTRY_3
         attributes:
-        - recursion_level: "512"
-        - FIM_MODE
+        - recursion_level: LEVEL_3
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_recursion_level/data/wazuh_recursion_windows_registry.yaml:25]: You are creating the registry keys with a 64 bit architecture, but the configuration does not set the architecture, being 32 bit by default. This causes the test to fail. Add an `arch` attribute to fix this.

**Review by jesusjimsa — CHANGES_REQUESTED** [review]
The last case is failing for me.
[recursion2.zip](https://github.com/wazuh/wazuh-qa/files/5464349/recursion2.zip)

**Review by jesusjimsa — CHANGES_REQUESTED** [review]
The test keeps failing in Windows.
[test_recursion.zip](https://github.com/wazuh/wazuh-qa/files/5523247/test_recursion.zip)

**Review by jesusjimsa — APPROVED** [review]
LGTM!

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**